### ISSUE TITLE
Returning null instead of empty collection for SCIMAttribute

### ISSUE THREAD

**Issue by jackychen-okta** [UnboundID/scim#5]
Hi, 

I notice that since a commit to fix the null strings, we return null instead of an empty collection. So if I do `group.getMembers()` on a GroupResource, it would be null and I can get a NPE on a foreach statement, e.g. `for (Collection<Entry<String>> entry : group.getMembers()`. 

http://stackoverflow.com/questions/6077909/is-there-a-way-to-avoid-null-check-before-the-for-each-loop-iteration-starts

I can do a null check, but it seems extraneous to do that before a foreach.

**Comment by coulbeck** [comment]
You have a good point - it’s somewhat burdensome and not very friendly to return null. That said, we are unlikely to change the behavior because we are working on an entirely new SCIM v2 client library.

**Comment by asaf** [comment]
@coulbeck out of the context of this issue, what's the state of the v2 client library ? is it going to be public under the same license, and if so, when?

Thank you for your great effort !

**Comment by coulbeck** [comment]
@asaf Currently it's planned to be public under the same license, sometime over summer (northern hemisphere!). Sorry I can't be any more specific right now.

**Comment by jwilleke** [comment]
So can the v2 client library work against PINGFederate? (Which appears to support SCIM v1.1)

**Comment by richardcardona** [comment]
No, the SCIM 2 standard is not backward-compatible with SCIM 1.x hence two different libraries.

**Comment by kenneth-ping** [comment]
This project has been archived and will no longer be maintained.  Please migrate your apps to SCIM v2.
